MM2RandoLib/Extensions/StringExtensions.cs

```diff
@@ -10,6 +11,17 @@ public static class StringExtensions
         //
         // Public Methods
         //
+        public static String TrimNonAlphanumeric(this String in_String)
```

Comment:
I think we should trim leading and trailing whitespace, but I don't see any harm in the space character (not tabs and other whitespace) appearing in the middle. You can use String `.Trim()` to get rid of leading and trailing whitespace. Then add ` ` to the list of characters to allow, so like `@"[^a-zA-Z0-9 ]"`.

Reply:
sure, we can do that.